src/utils.tsx

```diff
@@ -19,7 +19,7 @@ function useLocalStorageState<TState>(
     deserialize = JSON.parse,
   }: UseLocalStorageOptions<TState> = {},
 ) {
-  const [state, setState] = React.useState(() => {
+  const [state, setState] = React.useState<TState>(() => {
```

Comment:
Thanks! However we don't need this as it's correctly inferred based on the callback function:

![image](https://user-images.githubusercontent.com/1500684/126539416-ed7260c4-42ed-441c-a94a-c0e41fb01835.png)

Thanks anyway :)

Reply:
You are right! I was digging into why it didn't work for me and I found my error.

In my case I always use JSON.parse and JSON.stringify without allowing custom de-/serialization.
Hence JSON.parse returns any it messes up the types of the callback function.

Thanks for your explanation!

Reply:
Probably not a bad idea to be more explicit so I may decide to add your change in the future.